kernel/tests/golden_tables.rs

```diff
@@ -0,0 +1,343 @@
+                let table_path = url::Url::from_directory_path(table_path).unwrap();
+                let engine =
+                    DefaultEngine::try_new(
+                        &table_path,
+                        std::iter::empty::<(&str, &str)>(),
+                        Arc::new(TokioBackgroundExecutor::new()),
+                    ).unwrap();
+                let table = Table::new(table_path);
+                $test_fn(engine, table);
+                Ok(())
+            }
+        }
+    };
+}
+
+fn check_version_is_one(engine: DefaultEngine<TokioBackgroundExecutor>, table: Table) {
+    // assert latest version is 1
+    // TODO and there are no AddFiles
+    let snapshot = table.snapshot(&engine, None).unwrap();
+    assert_eq!(snapshot.version(), 1);
+}
+
+// All the test cases are below. Four test cases are currently supported:
+// 1. full_scan! - run the test with the latest snapshot
+// 2. negative_test! - run the test with the latest snapshot and expect it to fail
+// 3. golden_test! - run the test with the latest snapshot and a custom test function
+// 4. skip_test! - skip the test with a reason
+
+full_scan_test!("124-decimal-decode-bug");
+full_scan_test!("125-iterator-bug");
+full_scan_test!("basic-decimal-table");
+full_scan_test!("basic-decimal-table-legacy");
+full_scan_test!("basic-with-inserts-deletes-checkpoint");
+full_scan_test!("basic-with-inserts-merge");
+full_scan_test!("basic-with-inserts-overwrite-restore");
+full_scan_test!("basic-with-inserts-updates");
+full_scan_test!("basic-with-vacuum-protocol-check-feature");
+skip_test!("checkpoint": "test not yet implmented");
+skip_test!("corrupted-last-checkpoint-kernel": "BUG: should fallback to old commits/checkpoint");
+skip_test!("data-reader-array-complex-objects": "list field expected name item but got name element");
+skip_test!("data-reader-array-primitives": "list field expected name item but got name element");
+full_scan_test!("data-reader-date-types-America");
+full_scan_test!("data-reader-date-types-Asia");
+full_scan_test!("data-reader-date-types-Etc");
+full_scan_test!("data-reader-date-types-Iceland");
+full_scan_test!("data-reader-date-types-Jst");
+full_scan_test!("data-reader-date-types-Pst");
+full_scan_test!("data-reader-date-types-utc");
+full_scan_test!("data-reader-escaped-chars");
+skip_test!("data-reader-map": "map field named 'entries' vs 'key_value'");
+full_scan_test!("data-reader-nested-struct");
+skip_test!("data-reader-nullable-field-invalid-schema-key":
+  "list field expected name item but got name element");
+skip_test!("data-reader-partition-values": "list field expected name item but got name element");
+full_scan_test!("data-reader-primitives");
+full_scan_test!("data-reader-timestamp_ntz");
+skip_test!("data-reader-timestamp_ntz-id-mode": "id column mapping mode not supported");
+full_scan_test!("data-reader-timestamp_ntz-name-mode");
+
+// TODO test with predicate
```

Comment:
can you elaborate a bit on what exactly we need to do here?

Reply:
same as comment above I think i'll be able to implement these - but will size how much work is needed

Reply:
actually going to just punt on these for a separate PR just so we can get this merged and iterate